### PR TITLE
Expand robots.txt for Kedro-Viz and Kedro-Datasets docs

### PR DESCRIPTION
**Pull request by DimedS**
This PR updates the `robots.txt` file to include explicit allowances for the indexing of the Kedro-Viz and Kedro-Datasets documentation sections by search engines.

Changes Made:

- Added Allow: /projects/kedro-viz/en/stable/ to permit the indexing of the stable version of Kedro-Viz documentation.
- Added Allow: /projects/kedro-datasets/en/kedro-datasets-2.1.0/ to permit the indexing of the Kedro-Datasets documentation for version 2.1.0.

Previously, only the kedro project documentation was allowed for indexing, as specified in our robots.txt. By updating this file, we making it easier for users to find information about Kedro-Viz and Kedro-Datasets directly via search engines.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by noklam — COMMENTED** [review]
Is this ready to be reviewed? How should this be tested.

https://www.google.com/search?q=kedro+docs+configloader&oq=kedro+docs+configloader&gs_lcrp=EgZjaHJvbWUyCQgAEEUYORifBTIGCAEQRRg8MgkIAhAhGAoYoAEyCQgDECEYChigATIJCAQQIRgKGKAB0gEIMzk0NGowajmoAgCwAgA&client=ms-android-google&sourceid=chrome-mobile&ie=UTF-8

I see 0.18.1 docs on Google search so I suspect something isn't working properly.

**Comment by DimedS** [comment]
> Is this ready to be reviewed? How should this be tested.
> 
> https://www.google.com/search?q=kedro+docs+configloader&oq=kedro+docs+configloader&gs_lcrp=EgZjaHJvbWUyCQgAEEUYORifBTIGCAEQRRg8MgkIAhAhGAoYoAEyCQgDECEYChigATIJCAQQIRgKGKAB0gEIMzk0NGowajmoAgCwAgA&client=ms-android-google&sourceid=chrome-mobile&ie=UTF-8
> 
> I see 0.18.1 docs on Google search so I suspect something isn't working properly.

Yes, it's ready for review. Overall, it seems to function properly, though we are experiencing some issues specifically with Google—I'm not sure of the exact cause. However, it's mostly fine with other search engines; please see the screenshots from Yahoo, Yandex, and DuckDuckGo for reference. I believe it's impossible to test it before merging, as search engines will need some time to recognize the updated `robots.txt` and reflect those changes in their search results.
Maybe we need to do something special with google.
Screenshot 2024-03-22 at 12 04 23
Screenshot 2024-03-22 at 12 04 33
Screenshot 2024-03-22 at 12 04 51

File: docs/source/robots.txt [hunk 1]
```diff
@@ -7,3 +7,5 @@ Allow: /en/0.19.1/
 Allow: /en/0.19.0/
 Allow: /en/0.18.14/
 Allow: /en/0.17.7/
+Allow: /projects/kedro-viz/en/stable/
+Allow: /projects/kedro-datasets/en/kedro-datasets-2.1.0/
```
Comment (datajoely) [inline docs/source/robots.txt:11]: what about `latest` so at least it's dynamic?

Comment (astrojuanlu) [reply docs/source/robots.txt:11]: Yeah I agree we should be consistent in this file and continue to specify versions we allow, in this case

```
https://docs.kedro.org/projects/kedro-viz/en/v8.0.1/
```

@datajoely we don't want folks to land on `/latest` docs that represent unreleased versions

**Comment by astrojuanlu** [comment]
> we are experiencing some issues specifically with Google—I'm not sure of the exact cause

Interesting, might be related to #3708 somehow?

**Review by astrojuanlu — APPROVED** [review]
I suggested 1 modification but overall I am okay with this quick and dirty approach 👍🏽 

Longer term we should re-evaluate our policy in `robots.txt`. Originally it was done so that very old versions are indexed by search engines, but not shown on the Read the Docs flyout menu. But nowadays I'm of the opinion that maybe only the most recent `stable` version should be indexed. We don't have to make that change right now though.

Screenshot 2024-03-25 at 16 59 44

**Review by tynandebold — APPROVED** [review]
This looks good to me.

File: docs/source/robots.txt (hunk 1 above)
Comment (astrojuanlu) [reply docs/source/robots.txt:11]: Actually it's not `v8.0.1` _instead_ of `stable`, but _on top of_ (@DimedS pointed out to me in private)

So we'd need to have

viz/stable
viz/8.0.1
datasets/stable
datasets/2.1.0

Does that make sense?

Comment (astrojuanlu) [reply docs/source/robots.txt:11]: Opened #3741 to continue discussing about this

**Comment by noklam** [comment]
```
User-agent: *
Disallow: /
Allow: /en/stable/
Allow: /en/0.19.3/
Allow: /en/0.19.2/
Allow: /en/0.19.1/
Allow: /en/0.19.0/
Allow: /en/0.18.14/
Allow: /en/0.17.7/
```

For the reference, the robots.txt is still not updated. https://docs.kedro.org/robots.txt

**Comment by astrojuanlu** [comment]
@noklam it will only get updated in the next release, I believe

**Pull request merged**